WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -69,6 +62,15 @@ def index
       @competitions = @competitions.select { |competition| competition.contains?(params[:search]) }
     end
 
+    unless params[:event_ids].empty?
+      @competitions = @competitions.select do |competition|
+        (competition.eventSpecs.split & params[:event_ids]).any?
+      end.sort_by do |competition|
+        -(competition.eventSpecs.split & params[:event_ids]).count
+      end
+      # Note: If both competitions have the same count of matched events, they are still ordered by start date
```

Comment:
Oh, it's already tested. I've added this to ensure that nobody will get confused.